tests/api_tests/test_search.py

```diff
@@ -315,6 +317,32 @@ def test_custom_search_results(self):
         custom_score = custom_res["hits"][0]["_score"]
 
         self.assertEqual(custom_score, original_score)
+    
+    def test_opensearch_search_retry(self):
+        """
+            Ensures that retries are implemented due to transient
+            network errors occuring when sending search requests to OpenSearch.
+        """
+        self.client.create_index(self.index_name_1)
+        utilities.control_marqo_os("marqo-os", "stop")
+        
+        for search_method in ["TENSOR", "LEXICAL"]:
```

Comment:
Won't this start the second loop with marqo-os already started? Maybe move stop command inside the loop.

Also, It would be a good idea to always restart marqo-os in test tearDown. If used carelessly, this test could leave marqo-os off and make succeeding tests fail.

Reply:
good catch!